### PR TITLE
ncm-metaconfig: schema for slurm

### PR DESCRIPTION
**Pull request by itkovian**
Provides configuration support for the slurm scheduler

### CODE REVIEW

File: ncm-metaconfig/src/main/metaconfig/slurm/pan/schema.pan [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+    'health' ? slurm_conf_health_check
+    'timers' ? slurm_conf_timers
+    'prepilogue' ? slurm_conf_prolog_epilog
+    'scheduling' : slurm_conf_scheduling
+    'priority' : slurm_conf_job_priority
+    'accounting' : slurm_conf_accounting
+    'logging' : slurm_conf_logging
+    'power' ? slurm_conf_power
+    @{key is used as nodename, unless NodeName attribute is set}
+    'nodes' : slurm_conf_compute_nodes{}
```
[the remark this answers is not in the document]
Comment (itkovian) [reply ncm-metaconfig/src/main/metaconfig/slurm/pan/schema.pan]: Like so

~~~~
type slurm_nodes = {
    'compute' ? slurm_nodes_compute
    'down' ? slurm_nodes_down
};
~~~~

[the remark this answers is not in the document]
Comment (stdweird) [reply ncm-metaconfig/src/main/metaconfig/slurm/pan/schema.pan]: i have all commits already lined up. PR soonish

File: ncm-metaconfig/src/main/metaconfig/slurm/pan/schema.pan [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+    'timers' ? slurm_conf_timers
+    'prepilogue' ? slurm_conf_prolog_epilog
+    'scheduling' : slurm_conf_scheduling
+    'priority' : slurm_conf_job_priority
+    'accounting' : slurm_conf_accounting
+    'logging' : slurm_conf_logging
+    'power' ? slurm_conf_power
+    'nodes' ? slurm_conf_nodes
+    @{key is used as PartitionName, unless PartitionName attribute is set}
+    'partitions' : slurm_conf_partition{}
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/slurm/pan/schema.pan]: @itkovian this is optional, eg on submit-only hosts

**Comment by stdweird** [comment]
retest this please

**Comment by jrha** [comment]
@drossy might be interested in this

File: ncm-metaconfig/src/main/metaconfig/slurm/pan/config.pan [hunk 3]
```diff
@@ -0,0 +1,18 @@
+unique template metaconfig/slurm/config;
+
+@{when defined, indicate this is a master. when false, this is a node.
```
[the remark this answers is not in the document]
Comment (jrha) [reply ncm-metaconfig/src/main/metaconfig/slurm/pan/config.pan]: It looks like it's just the comment that is wrong...

File: ncm-metaconfig/src/main/metaconfig/slurm/pan/acct_gather.pan [hunk 4]
```diff
@@ -0,0 +1,19 @@
+unique template metaconfig/slurm/acct_gather;
+
+include 'components/metaconfig/config';
+include 'metaconfig/slurm/schema';
+
+bind "/software/components/metaconfig/services/{/etc/slurm/acct_gather.conf}/contents" = slurm_acct_gather_conf;
+
+variable SLURM_DAEMONS = {
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/slurm/pan/acct_gather.pan]: this variable egenration should be moved to a separate template and included where needed

File: ncm-metaconfig/src/main/metaconfig/slurm/pan/config.pan [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  leave undef when this is e.g. a loginnode}
+variable SLURM_IS_MASTER ?= undef;
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/slurm/pan/config.pan]: @itkovian are these still used anywhere?

**Comment by jrha** [comment]
@itkovian any update on @stdweird's comments?

**Comment by stdweird** [comment]
test this please

**Pull request merged**